### ISSUE TITLE
Data spreadsheet migration fixes

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-cloud-cognitive#2875]
From the [data spreadsheet](https://github.ibm.com/CDAI-design/pal/issues/1157) migration review —

### Design specs
- [ ] In dark themes, selected headers or rows are not readable — should update theme token
  > ![image](https://user-images.githubusercontent.com/1717398/232153121-eb89411b-5a22-4a16-874d-56b42060bfe7.png)

- [ ] Row headers use opacity not in 1 that causes overlapping text.
  > ![image](https://user-images.githubusercontent.com/1717398/232154498-081de381-7285-41cb-8bc9-2c5cd476a6a2.png)

- [ ] Dark mode seems to use blue instead of cyan even with `theme` set
